apollo-router/src/services/subgraph.rs

```diff
@@ -39,6 +39,8 @@ pub struct Request {
 
     pub context: Context,
 
+    /// Name of the subgraph, it's an Option to not introduce breaking change
+    pub(crate) subgraph_name: Option<String>,
```

Comment:
This struct is `non-exhaustive` so does this need to be an `Option`?

Reply:
Hmm not for the struct itself but for the builder. What am I supposed to put in the builder ? If I'm not using an option it will break all the existing builders because it will require the subgraph_name. And so if it's not an option, I will still have to use an Option in the builder

Reply:
Does that mean `buildstructor` means there's no point marking a struct as `non-exhaustive`?

Reply:
I think so... Unless I'm missing something ?! cc @BrynCooke

Reply:
We could default the name in the builder if it is not supplied. That way in the struct is can be mandatory.

Reply:
That seems nicer.
Not this PR: but does it mean we should put comments on any other structs which are marked `non-exhaustive` and use `buildstructor` so future us know how to handle this situation? We could raise an issue for that.